api/enki.py

```diff
@@ -0,0 +1,633 @@
+        else:
+            # This should only happen during a test.
+            self.metadata_wrangler = None
+        self.api = api or EnkiAPI.from_environment(self._db)
+        self.bibliographic_coverage_provider = (
+            EnkiBibliographicCoverageProvider(self._db, enki_api=api)
+        )
+
+    def run(self):
+        super(EnkiImport, self).run()
+
+    def run_once(self, start, cutoff):
+        # Give us five minutes of overlap because it's very important
+        # we don't miss anything.
+        since = start-self.FIVE_MINUTES
+        x=0
+        step=2000
+        while x < 80000:
+            availability = self.api.availability(since=since, strt=x, qty=step)
+	    status_code = availability.status_code
+            content = availability.content
+            count = 0
+            for bibliographic, circulation in BibliographicParser().process_all(
+                    content, "//enki:title"):
+                self.process_book(bibliographic, circulation)
+                count += 1
+                if count % self.batch_size == 0:
+                    self._db.commit()
+            x += step
+
+    def process_book(self, bibliographic, availability):
+
+        license_pool, new_license_pool = availability.license_pool(self._db)
+        edition, new_edition = bibliographic.edition(self._db)
+        license_pool.edition = edition
+        policy = ReplacementPolicy(
+            identifiers=False,
+            subjects=True,
+            contributions=True,
+            formats=True,
+        )
+        availability.apply(
+            pool=license_pool,
+            replace=policy,
+        )
+        if new_edition:
+            bibliographic.apply(edition, replace=policy)
+
+        if new_license_pool or new_edition:
+            # At this point we have done work equivalent to that done by
+            # the EnkiBibliographicCoverageProvider. Register that the
+            # work has been done so we don't have to do it again.
+            identifier = edition.primary_identifier
+            self.bibliographic_coverage_provider.handle_success(identifier)
+            self.bibliographic_coverage_provider.add_coverage_record_for(
+                identifier
+            )
+
+        return edition, license_pool
+
+#Copied from 3M. Eventually we might want to refactor
+class EnkiCollectionReaper(IdentifierSweepMonitor):
+    """Check for books that are in the local collection but have left the Enki collection."""
+    def __init__(self, _db, api=None, interval_seconds=3600*4):
+        super(EnkiCollectionReaper, self).__init__(_db, "Enki Collection Reaper", interval_seconds)
+        self._db = _db
+        if not api:
+            api = EnkiAPI.from_environment(_db)
+        self.api = api
+        self.data_source = DataSource.lookup(self._db, DataSource.ENKI)
+
+    def run(self):
+        self.api = EnkiAPI.from_environment(self._db)
```

Comment:
You shouldn't need to set self.api again--it was set in the constructor, potentially to a mocked value.